### PR TITLE
Fixing build errors

### PR DESCRIPTION
**Pull request by bpandey-CS**
Solutions explored (that work):
1. Custom Dockerfile (for `spark-base` image) to remove `pip-python` installation, works but we would need to host it and it would need some maneuvering while deploying the skill-template (probably would have to handle in the scripts).
2. Upgrade the spark version and its dependencies `hadoop-aws`, `aws-java-sdk-bundle` to a compatible version and update the configuration because of the upgrade.

Went with the 2nd solution. upgrade in `cortex-python` version needed some code changes, tested with a model trained in the current version of sklearn (a mismatch in versions for training and prediction throws a warning)

Vuln checks:
Attaching the snyk reports here for the built images (1 critical ([glibc](https://security.snyk.io/vuln/SNYK-DEBIAN11-GLIBC-1296898)) 35 low)
[spark_cotainer_scan.txt](https://github.com/CognitiveScale/cortex-fabric-examples/files/7737765/spark_cotainer_scan.txt)

[spark_batch_predict_scan.txt](https://github.com/CognitiveScale/cortex-fabric-examples/files/7737763/spark_batch_predict_scan.txt)

Did a manual scan for JndiLookup class in the images for the log4j-core vuln, didn't find anything (`unzip -l log4j-1.2.17.jar | grep -i "JndiLookup"`)

### CODE REVIEW

**Comment by svangapally-cs** [comment]
LGTM as long as we are able to invoke the skill without any issues.

**Review by svangapally-cs — APPROVED** [review]

**Comment by pkandarpa-cs** [comment]
@svangapally-cs we should link Anthem to this new folder in fabric-examples for sample intervention skills

File: spark-batch-prediction/spark-base/build-spark-image.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-spark_version=3.0.2
+spark_version=3.2.0
```
Comment (pkandarpa-cs) [inline spark-batch-prediction/spark-base/build-spark-image.sh:3]: did you test with this minor version upgrade?

File: spark-batch-prediction/src/main/python/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-pandas==1.0.5
-cortex-python==6.0.1
+pandas==1.2.
+cortex-python==6.1.0
 catboost==0.25.1
-scikit-learn==0.22.1
\ No newline at end of file
+scikit-learn==1.0.1
```
Comment (pkandarpa-cs) [inline spark-batch-prediction/src/main/python/requirements.txt:4]: this is a major version change. How safe is it?

File: spark-batch-prediction/spark-base/build-spark-image.sh (hunk 1 above)
Comment (bpandey-CS) [reply spark-batch-prediction/spark-base/build-spark-image.sh:3]: yes, tested and works

**Comment by pkandarpa-cs** [comment]
Any remediations for the critical
Vulnerability?

**Pull request merged**